### DISCUSSION TITLE
How do I use plugins with remark-highlight.js?

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [remarkjs/remark#701] [Q&A] [answered]
Looking at: https://github.com/remarkjs/remark-highlight.js
I'm wanting to use: https://github.com/NullVoxPopuli/highlightjs-glimmer

I also want to alias a language tag `sh-session` -> `sh` or `bash` (auto-generated from a CLI tool that I don't control the output of).
Is that possible as well?

**Comment by wooorm** [comment, marked as answer]
You probably should use [`rehype-highlight`](https://github.com/rehypejs/rehype-highlight), as noted in the `remark-highlight.js` readme. It has support for aliases and registering new languages (which I believe glimmer is?)

**Reply by NullVoxPopuli** [reply]
Thanks, I'll give it a go! 
(And yeah, glimmer is a new language -- couldn't pr to hljs, because I needed extra shipped or module formats from what they offer)

**Reply by NullVoxPopuli** [reply]
I made this test: https://github.com/NullVoxPopuli/highlightjs-glimmer/pull/35/files#diff-a14263a13ac66963bbbbe860d2028c6ea37cadba53cd5e8b2e7e08c2f0174713

but get an error: https://github.com/NullVoxPopuli/highlightjs-glimmer/pull/35/checks?check_run_id=2479092663#step:6:12

**Reply by wooorm** [reply]
The error seems to be related to highlight.js.

Is your `setup` function a [function such as this](https://github.com/highlightjs/highlight.js/blob/02fab278568ae059a7765766f211f9f4770962f7/src/languages/xml.js#L11)?

**Reply by NullVoxPopuli** [reply]
It is: https://github.com/NullVoxPopuli/highlightjs-glimmer/blob/main/src/index.js#L5

**Reply by NullVoxPopuli** [reply]
But I get errors no matter which language I specify.

**Reply by NullVoxPopuli** [reply]
I opened a failing test: https://github.com/rehypejs/rehype-highlight/pull/16
